prometheus/exporter/watchdog.py

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/python
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import subprocess
+import json
+import sys
+import requests
+import logging  
+from logging.handlers import RotatingFileHandler
+import time
+
+logger = logging.getLogger("watchdog")  
+
+def parse_pods_status(podsJsonObject, outputFile):
+    # metrics 
+    kube_pod_status_probe_not_ready = 0
+    kube_pod_status_phase_failed = 0
+    kube_pod_status_phase_unknown = 0
+    pod_container_status_waiting = 0
+    pod_container_status_terminated = 0
+    pod_container_status_not_ready = 0
+    pod_container_status_restarted_pod_count = 0 # if one pod restarts > 1, add 1 to this metrics
+
```

Comment:
Suggest we group this information by controllers (ds, deployments, etc.), and in each log, we include controller name, total pods also.

Reply:
added